app/models/rook.rb

```diff
@@ -1,3 +1,26 @@
 class Rook < Piece
 
+	def valid_move?(dest_x, dest_y)
+
+		#a move can not be a non-move
+    	return false if dest_x == self.x_position && dest_y == self.y_position
+
+		#no obstructions allowed
+		return false if is_obstructed?(dest_x, dest_y) == true
```

Comment:
Yep, I erroneously thought that a controller (or the piece model) could process every piece's move through `is_obstructed?`; but I forgot that some pieces (kings, pawns, and knights) are not supposed to use `is_obstructed?` 'cuz it's not a part of their movesets. Brain fart on my end. 😅